libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -322,17 +333,13 @@ void ReferencesResolver::endVisit(VariableDeclaration const& _variable)
 			type = ref->copyForLocation(typeLoc, isPointer);
 		}
 		else if (varLoc != Location::Default && !ref)
-			fatalTypeError(_variable.location(), "Storage location can only be given for array or struct types.");
-
-		if (!type)
-			fatalTypeError(_variable.location(), "Invalid type name.");
+			typeError(_variable.location(), "Storage location can only be given for array or struct types.");
 
+		_variable.annotation().type = type;
 	}
 	else if (!_variable.canHaveAutoType())
-		fatalTypeError(_variable.location(), "Explicit type needed.");
+		typeError(_variable.location(), "Explicit type needed.");
 	// otherwise we have a "var"-declaration whose type is resolved by the first assignment
-
-	_variable.annotation().type = type;
```

Comment:
This is not assigned anymore in the final `else` case. Is that intended?

Reply:
Yes, `type` is only guaranteed to have a value assigned in the main branch.